lib/node_modules/@stdlib/math/base/special/gammasgn/test/test.js

```diff
@@ -0,0 +1,165 @@
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var tape = require( 'tape' );
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var isPositiveZero = require( '@stdlib/math/base/assert/is-positive-zero' );
+var EPS = require( '@stdlib/constants/float64/eps' );
+var abs = require( '@stdlib/math/base/special/abs' );
+var gammasgn = require( './../lib' );
+
+
+// FIXTURES //
+
+var random = require( './fixtures/python/random.json' );
+
+var mediumNegative = require( './fixtures/python/medium_negative.json' );
+var mediumPositive = require( './fixtures/python/medium_positive.json' );
+var smallPositive = require( './fixtures/python/small_positive.json' );
+var smallNegative = require( './fixtures/python/small_negative.json' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.equal( typeof gammasgn, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function computes the sign of `gamma` evaluated in `x`', function test( t ) {
+	var expected;
+	var delta;
+	var tol;
+	var x;
+	var v;
+	var i;
+
+	x = random.x;
+	expected = random.expected;
+
+	for ( i = 0; i < x.length; i++ ) {
+		v = gammasgn( x[ i ] );
+		delta = abs( v - expected[ i ] );
+		tol = EPS * abs( expected[ i ] );
+		t.ok( delta <= tol, 'within tolerance. x: ' + x[ i ] + '. Value: ' + v + '. Expected: ' + expected[ i ] + '. Delta: ' + delta + '. Tolerance: ' + tol + '.' );
+	}
+	t.end();
+});
+
+tape( 'the function computes the sign of `gamma` for small positive numbers', function test( t ) {
+	var expected;
+	var delta;
+	var tol;
+	var x;
+	var v;
+	var i;
+
+	x = smallPositive.x;
+	expected = smallPositive.expected;
+
+	for ( i = 0; i < x.length; i++ ) {
+		v = gammasgn( x[ i ] );
+		delta = abs( v - expected[ i ] );
+		tol = EPS * abs( expected[ i ] );
+		t.ok( delta <= tol, 'within tolerance. x: ' + x[ i ] + '. Value: ' + v + '. Expected: ' + expected[ i ] + '. Delta: ' + delta + '. Tolerance: ' + tol + '.' );
+	}
+	t.end();
+});
+
+tape( 'the function computes the sign of `gamma` for medium positive numbers', function test( t ) {
+	var expected;
+	var delta;
+	var tol;
+	var x;
+	var v;
+	var i;
+
+	x = mediumPositive.x;
+	expected = mediumPositive.expected;
+
+	for ( i = 0; i < x.length; i++ ) {
+		v = gammasgn( x[ i ] );
+		delta = abs( v - expected[ i ] );
+		tol = EPS * abs( expected[ i ] );
+		t.ok( delta <= tol, 'within tolerance. x: ' + x[ i ] + '. Value: ' + v + '. Expected: ' + expected[ i ] + '. Delta: ' + delta + '. Tolerance: ' + tol + '.' );
+	}
+	t.end();
+});
+
+tape( 'the function computes the sign of `gamma` for small negative numbers', function test( t ) {
+	var expected;
+	var delta;
+	var tol;
+	var x;
+	var v;
+	var i;
+
+	x = smallNegative.x;
+	expected = smallNegative.expected;
+
+	for ( i = 0; i < x.length; i++ ) {
+		v = gammasgn( x[ i ] );
```

Comment:
Same comment as above.

We only need to perform approx floating-point comparison when values are subject to floating-point rounding error.